waldenn/conzept
# Adding Docker support to The project with SSL certificate generation.

This pull request introduces Docker support for the repository, making it easier to deploy and manage. The following changes have been made:

- Added Dockerfile to build the Docker image for the application.
- Created docker-compose.yml to manage the services and dependencies required for the application.- 
- Integrated Certbot service to generate and manage SSL certificates using Let's Encrypt.
- Implemented init_https.sh script that automates the process of obtaining an SSL certificate for the domain.
- Included a sample nginx.conf file with the necessary configuration for serving the application over HTTPS.
These changes allow users to quickly set up and run the application in a Docker environment, with HTTPS enabled for added security.

To test this PR, follow these steps:

1. Ensure Docker and Docker Compose are installed on your system following the script inside the ***scripts*** folder
2. Clone this repository and checkout this branch.
3. Update the settings.conf script with your domain information.

4. Run docker-compose build to build the Docker images.
5. Execute ./init_https.sh to generate an SSL certificate for your domain.
6. Run docker-compose to start the application and its dependencies.
7. Access your application via your domain with HTTPS.
Please review the changes and let me know if you have any questions or suggestions for improvement.
